src/App.tsx

```diff
@@ -21,75 +21,126 @@ enum SortType {
   LENGTH,
 }
 
+function getSortChoice(wordOne: string, wordTwo: string, SortType2: SortType) {
+  switch (+SortType2) {
+    case SortType.ALPHABET:
+      return wordOne.localeCompare(wordTwo);
+    case SortType.LENGTH:
+      return wordOne.length - wordTwo.length;
+    case SortType.NONE:
+    default:
+      return 0;
+  }
+}
+
 type ReorderOptions = {
   sortType: SortType,
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort(
+    (
+      wordOne, wordTwo,
+    ) => getSortChoice(
+      wordOne, wordTwo, sortType,
+    ),
+  );
+
+  if (isReversed === true) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component <{}, State> {
+  state = {
+    isReversed: false,
+    sortType: 0,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ isReversed: false, sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ isReversed: false, sortType: SortType.LENGTH });
```

Comment:
do not set `isReversed: false` - according to requirements it should be active (check [working example](https://mate-academy.github.io/react_list-of-goods/)):
![image](https://github.com/mate-academy/react_list-of-goods/assets/86923089/c5161c5a-5d7f-4f20-9a8d-fe9b0c548e7d)